subsys/net/lib/sockets/sockets.c

```diff
@@ -329,6 +376,10 @@ int z_impl_zsock_shutdown(int sock, int how)
 
 	k_mutex_unlock(lock);
 
+#if defined(CONFIG_NET_SOCKETS_OBJ_CORE)
+	(void)sock_obj_core_dealloc(sock);
```

Comment:
Should dealloc really take place in `shutdown`? The socket still remains open after this call (i. e. you still need to `close()` it, where the dealloc is called).

Reply:
Good point, it should not be here. Will fix.